main/src/main/java/org/mobilitydata/gtfsvalidator/report/JsonReportSummary.java

```diff
@@ -0,0 +1,75 @@
+package org.mobilitydata.gtfsvalidator.report;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.mobilitydata.gtfsvalidator.report.model.AgencyMetadata;
+import org.mobilitydata.gtfsvalidator.report.model.FeedMetadata;
+import org.mobilitydata.gtfsvalidator.runner.ValidationRunnerConfig;
+import org.mobilitydata.gtfsvalidator.util.VersionInfo;
+
+/**
+ * Used to generate the summary of json report using the built-in class serialization of GSON.
+ * Fields names used here will be the same as in the generated json file.
+ */
+@SuppressWarnings("unused") // The fields of this class are only read by Gson when serializing.
+public class JsonReportSummary {
```

Comment:
Hey, I had a macro comment on this class.

This class defines the exported API for the JSON report summary.  As such, identifiers used here should be stable, since folks external users will be depending on this API.  However, this summary depends on a number of identifiers export from `FeedMetadata` and `AgencyMetadata`, two classes that were written primarily as UI presentation model classes.  As such, I think there is a risk someone might make a change to one of those UI presentation classes, not realizing it would impact the exported JSON api.

For example, the key names in `feedInfo` below, any of the data in `AgencyMetadata`, the keys in `counts`, and the identifiers in `gtfsComponents`.  They'd all ideally be defined fields or enums, where it's very clear that changing a value would change the exported API.

Does that feedback make sense?

Reply:
Created classes devoted to the json generation with controlled fields and copied values from the existing classes. 
Also added some string constants used in the Html and Json generation code so it's obvious changing the string could have an effect in the Json report code.

Continued using gtfsComponents directly. We will have to be avoid changing the component names or document it properly.

Reply:
These changes look good.  Thanks!

Doesn't have to be in this PR, but long-term, I think it would be more valuable to have GTFS Components modeled as an enum internally, not a string.  It might ultimately be rendered as a string, but I think a enum would be more practical for long-term programatic access.